### PR TITLE
Make provenance printing optional

### PR DESCRIPTION
**Pull request by tavinathanson**
Also fix some docstrings.

@jburos

### CODE REVIEW

File: cohorts/load.py [hunk 1]
```diff
@@ -213,6 +213,14 @@ class Cohort(Collection):
     responder_pfs_equals_os : bool
         Ensure that the PFS values for responders (not progressed) are equal to
         OS values.
+    check_provenance : bool
+        Verify that the cached provenance is equal to the current environment.
+    print_provenance : bool
+        Print a cached provenance summary.
```
Comment (jburos) [inline cohorts/load.py]: Technically, this is a summary of the cache provenance .. not a cached summary of the provenance files. 
Maybe say "Print a summary of cache file provenance"?

**Comment by jburos** [comment]
Thanks @tavinathanson - just one minor suggestion, but otherwise looks good to me. Thanks for updating the docstrings

File: cohorts/load.py (hunk 1 above)
Comment (tavinathanson) [inline cohorts/load.py]: Was supposed to be read as (cached provenance) summary vs. cached (provenance summary), but I'll update :)

**Comment by jburos** [comment]
One comment - this makes me want to know, in which scenarios you'd like to turn off the data source summary? 

Please LMK (or maybe create issues) if there are modifications that should be made to that process (ie to check but not print, or to print but not warn, to exclude certain packages from the summary, etc).

**Comment by tavinathanson** [comment]
I have notebooks that initialize `Cohort`s inside a loop, and I don't want so many summaries printed. Open to smarter ways of handling this, though this seemed like a simple bandaid for starters. Will make an issue!

**Pull request merged**